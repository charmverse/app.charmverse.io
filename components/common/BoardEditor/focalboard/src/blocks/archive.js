"use strict";
exports.__esModule = true;
exports.ArchiveUtils = void 0;
var ArchiveUtils = /** @class */ (function () {
    function ArchiveUtils() {
    }
    ArchiveUtils.buildBlockArchive = function (blocks) {
        var header = {
            version: 1,
            date: Date.now()
        };
        var headerString = JSON.stringify(header);
        var content = "".concat(headerString, "\n");
        for (var _i = 0, blocks_1 = blocks; _i < blocks_1.length; _i++) {
            var block = blocks_1[_i];
            var line = {
                type: 'block',
                data: block
            };
            var lineString = JSON.stringify(line);
            content += lineString;
            content += '\n';
        }
        return content;
    };
    ArchiveUtils.parseBlockArchive = function (contents) {
        var blocks = [];
        var allLineStrings = contents.split('\n');
        if (allLineStrings.length >= 2) {
            var headerString = allLineStrings[0];
            var header = JSON.parse(headerString);
            if (header.date && header.version >= 1) {
                var lineStrings = allLineStrings.slice(1);
                var lineNum = 2;
                for (var _i = 0, lineStrings_1 = lineStrings; _i < lineStrings_1.length; _i++) {
                    var lineString = lineStrings_1[_i];
                    if (!lineString) {
                        // Ignore empty lines, e.g. last line
                        continue;
                    }
                    var line = JSON.parse(lineString);
                    if (!line || !line.type || !line.data) {
                        throw new Error("ERROR parsing line ".concat(lineNum));
                    }
                    switch (line.type) {
                        case 'block': {
                            var blockLine = line;
                            var block = blockLine.data;
                            blocks.push(block);
                            break;
                        }
                    }
                    lineNum += 1;
                }
            }
            else {
                throw new Error('ERROR parsing header');
            }
        }
        return blocks;
    };
    return ArchiveUtils;
}());
exports.ArchiveUtils = ArchiveUtils;
