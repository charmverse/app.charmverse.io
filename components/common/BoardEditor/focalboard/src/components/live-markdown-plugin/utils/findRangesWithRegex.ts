// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
const findRangesWithRegex = (text: string, regex: RegExp): number[][] => {
    const ranges: number[][] = []
    let matches

    do {
        matches = regex.exec(text)
        if (matches) {
            ranges.push([matches.index, (matches.index + matches[0].length) - 1])
        }
    } while (matches)

    return ranges
}

export default findRangesWithRegex
