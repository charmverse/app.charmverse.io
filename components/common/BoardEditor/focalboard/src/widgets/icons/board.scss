.BoardIcon {
    stroke-width: 8px;
}
