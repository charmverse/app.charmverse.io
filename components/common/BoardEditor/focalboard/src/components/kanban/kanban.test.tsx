// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {fireEvent, render, screen, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import React from 'react'
import {Provider as ReduxProvider} from 'react-redux'
import {mocked} from 'ts-jest/utils'
import userEvent from '@testing-library/user-event'

import {IPropertyOption, IPropertyTemplate} from '../../blocks/board'
import {TestBlockFactory} from '../../test/testBlockFactory'
import {mockDOM, mockStateStore, wrapDNDIntl} from '../../testUtils'
import {Utils} from '../../utils'
import {mutator} from '../../mutator'

import Kanban from './kanban'

global.fetch = jest.fn()
jest.mock('../../utils')
const mockedUtils = mocked(Utils, true)
const mockedchangePropertyOptionValue = jest.spyOn(mutator, 'changePropertyOptionValue')
const mockedChangeViewCardOrder = jest.spyOn(mutator, 'changeViewCardOrder')
const mockedinsertPropertyOption = jest.spyOn(mutator, 'insertPropertyOption')

describe('src/component/kanban/kanban', () => {
    const board = TestBlockFactory.createBoard()
    const activeView = TestBlockFactory.createBoardView(board)
    const card1 = TestBlockFactory.createCard(board)
    card1.id = 'id1'
    card1.fields.properties = {id: 'property_value_id_1'}
    const card2 = TestBlockFactory.createCard(board)
    card2.id = 'id2'
    card2.fields.properties = {id: 'property_value_id_1'}
    const card3 = TestBlockFactory.createCard(board)
    card3.id = 'id3'
    card3.fields.properties = {id: 'property_value_id_2'}
    activeView.fields.kanbanCalculations = {
        id1: {
            calculation: 'countEmpty',
            propertyId: '1',

        },
    }
    const optionQ1:IPropertyOption = {
        color: 'propColorOrange',
        id: 'property_value_id_1',
        value: 'Q1',
    }
    const optionQ2:IPropertyOption = {
        color: 'propColorBlue',
        id: 'property_value_id_2',
        value: 'Q2',
    }
    const optionQ3:IPropertyOption = {
        color: 'propColorDefault',
        id: 'property_value_id_3',
        value: 'Q3',
    }

    const groupProperty: IPropertyTemplate = {
        id: 'id',
        name: 'name',
        type: 'text',
        options: [optionQ1, optionQ2],
    }

    const state = {
        cards: {
            cards: [card1, card2, card3],
        },
        contents: {},
        comments: {
            comments: {},
        },
    }
    const store = mockStateStore([], state)
    beforeAll(() => {
        console.error = jest.fn()
        mockDOM()
    })
    beforeEach(jest.resetAllMocks)
    test('should match snapshot', () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2, card3]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))
        expect(container).toMatchSnapshot()
    })
    test('do not return a kanban with groupByProperty undefined', () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={undefined}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))

        expect(mockedUtils.assertFailure).toBeCalled()
        expect(container).toMatchSnapshot()
    })
    test('return kanban and drag card to other card ', async () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))

        const cardsElement = container.querySelectorAll('.KanbanCard')
        expect(cardsElement).not.toBeNull()
        expect(cardsElement).toHaveLength(3)
        fireEvent.dragStart(cardsElement[0])
        fireEvent.dragEnter(cardsElement[1])
        fireEvent.dragOver(cardsElement[1])
        fireEvent.drop(cardsElement[1])
        expect(mockedUtils.log).toBeCalled()

        await waitFor(async () => {
            expect(mockedChangeViewCardOrder).toBeCalled()
        })
    })
    test('return kanban and change card column', async () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))

        const cardsElement = container.querySelectorAll('.KanbanCard')
        expect(cardsElement).not.toBeNull()
        expect(cardsElement).toHaveLength(3)
        const columnQ2Element = container.querySelector('.octo-board-column:nth-child(2)')
        expect(columnQ2Element).toBeDefined()
        fireEvent.dragStart(cardsElement[0])
        fireEvent.dragEnter(columnQ2Element!)
        fireEvent.dragOver(columnQ2Element!)
        fireEvent.drop(columnQ2Element!)
        await waitFor(async () => {
            expect(mockedChangeViewCardOrder).toBeCalled()
        })
    })
    test('return kanban and change card column to hidden column', async () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))

        const cardsElement = container.querySelectorAll('.KanbanCard')
        expect(cardsElement).not.toBeNull()
        expect(cardsElement).toHaveLength(3)
        const columnQ3Element = container.querySelector('.octo-board-hidden-item')
        expect(columnQ3Element).toBeDefined()
        fireEvent.dragStart(cardsElement[0]!)
        fireEvent.dragEnter(columnQ3Element!)
        fireEvent.dragOver(columnQ3Element!)
        fireEvent.drop(columnQ3Element!)
        await waitFor(async () => {
            expect(mockedChangeViewCardOrder).toBeCalled()
        })
    })
    test('return kanban and click on New', () => {
        const mockedAddCard = jest.fn()
        render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={mockedAddCard}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))
        const allButtonsNew = screen.getAllByRole('button', {name: '+ New'})
        expect(allButtonsNew).not.toBeNull()
        userEvent.click(allButtonsNew[0])
        expect(mockedAddCard).toBeCalledTimes(1)
    })

    test('return kanban and click on KanbanCalculationMenu', () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))
        const buttonKanbanCalculation = screen.getByRole('button', {name: '2'})
        expect(buttonKanbanCalculation).toBeDefined()
        userEvent.click(buttonKanbanCalculation!)
        expect(container).toMatchSnapshot()
    })

    test('return kanban and change title on KanbanColumnHeader', async () => {
        const {container} = render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))

        const inputTitle = screen.getByRole('textbox', {name: optionQ1.value})
        expect(inputTitle).toBeDefined()
        fireEvent.change(inputTitle, {target: {value: ''}})
        userEvent.type(inputTitle, 'New Q1')
        fireEvent.blur(inputTitle)

        await waitFor(async () => {
            expect(mockedchangePropertyOptionValue).toBeCalledWith(board, groupProperty, optionQ1, 'New Q1')
        })

        expect(container).toMatchSnapshot()
    })
    test('return kanban and add a group', async () => {
        render(wrapDNDIntl(
            <ReduxProvider store={store}>
                <Kanban
                    board={board}
                    activeView={activeView}
                    cards={[card1, card2]}
                    groupByProperty={groupProperty}
                    visibleGroups={[
                        {
                            option: optionQ1,
                            cards: [card1, card2],
                        }, {
                            option: optionQ2,
                            cards: [card3],
                        },
                    ]}
                    hiddenGroups={[
                        {
                            option: optionQ3,
                            cards: [],
                        },
                    ]}
                    selectedCardIds={[]}
                    readonly={false}
                    onCardClicked={jest.fn()}
                    addCard={jest.fn()}
                    showCard={jest.fn()}
                />
            </ReduxProvider>,
        ))
        const buttonAddGroup = screen.getByRole('button', {name: '+ Add a group'})
        expect(buttonAddGroup).toBeDefined()
        userEvent.click(buttonAddGroup)
        await waitFor(() => {
            expect(mockedinsertPropertyOption).toBeCalled()
        })
    })
})
