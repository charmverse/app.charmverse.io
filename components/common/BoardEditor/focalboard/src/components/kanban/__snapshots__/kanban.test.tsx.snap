// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/kanban/kanban do not return a kanban with groupByProperty undefined 1`] = `
<div>
  <div />
</div>
`;

exports[`src/component/kanban/kanban return kanban and change title on KanbanColumnHeader 1`] = `
<div>
  <div
    class="Kanban"
  >
    <div
      class="octo-board-header"
      id="mainBoardHeader"
    >
      <div
        class="octo-board-header-cell KanbanColumnHeader"
        draggable="true"
        style="opacity: 1;"
      >
        <span
          class="Label propColorOrange "
        >
          <input
            class="Editable undefined"
            placeholder="New Select"
            spellcheck="true"
            title="New Q1"
            value="New Q1"
          />
        </span>
        <div
          class="KanbanCalculation"
        >
          <button
            title="2"
            type="button"
          >
            <span>
              2
            </span>
          </button>
        </div>
        <div
          class="octo-spacer"
        />
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="Button IconButton"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
        <button
          class="Button IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-plus AddIcon"
          />
        </button>
      </div>
      <div
        class="octo-board-header-cell KanbanColumnHeader"
        draggable="true"
        style="opacity: 1;"
      >
        <span
          class="Label propColorBlue "
        >
          <input
            class="Editable undefined"
            placeholder="New Select"
            spellcheck="true"
            title="Q2"
            value="Q2"
          />
        </span>
        <div
          class="KanbanCalculation"
        >
          <button
            title="1"
            type="button"
          >
            <span>
              1
            </span>
          </button>
        </div>
        <div
          class="octo-spacer"
        />
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="Button IconButton"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
        <button
          class="Button IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-plus AddIcon"
          />
        </button>
      </div>
      <div
        class="octo-board-header-cell narrow"
      >
        Hidden columns
      </div>
      <div
        class="octo-board-header-cell narrow"
      >
        <button
          type="button"
        >
          <span>
            + Add a group
          </span>
        </button>
      </div>
    </div>
    <div
      class="octo-board-body"
      id="mainBoardBody"
    >
      <div
        class="octo-board-column"
      >
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <button
          type="button"
        >
          <span>
            + New
          </span>
        </button>
      </div>
      <div
        class="octo-board-column"
      >
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <button
          type="button"
        >
          <span>
            + New
          </span>
        </button>
      </div>
      <div
        class="octo-board-column narrow"
      >
        <div
          class="octo-board-hidden-item"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper"
            role="button"
          >
            <span
              class="Label propColorDefault "
            >
              Q3
            </span>
          </div>
          <button
            type="button"
          >
            <span>
              0
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/component/kanban/kanban return kanban and click on KanbanCalculationMenu 1`] = `
<div>
  <div
    class="Kanban"
  >
    <div
      class="octo-board-header"
      id="mainBoardHeader"
    >
      <div
        class="octo-board-header-cell KanbanColumnHeader"
        draggable="true"
        style="opacity: 1;"
      >
        <span
          class="Label propColorOrange "
        >
          <input
            class="Editable undefined"
            placeholder="New Select"
            spellcheck="true"
            title="Q1"
            value="Q1"
          />
        </span>
        <div
          class="KanbanCalculation"
        >
          <button
            title="2"
            type="button"
          >
            <span>
              2
            </span>
          </button>
          <div
            class="CalculationOptions css-2b097c-container"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            >
              <span
                id="aria-selection"
              />
              <span
                id="aria-context"
              >
                 7 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.
              </span>
            </span>
            <div
              class="CalculationOptions__control CalculationOptions__control--is-focused CalculationOptions__control--menu-is-open css-1s59geg-Control"
            >
              <div
                class="CalculationOptions__value-container CalculationOptions__value-container--has-value css-1mxrbau-ValueContainer"
              >
                <div
                  class="CalculationOptions__single-value css-1brck82-singleValue"
                >
                  Count
                </div>
                <input
                  aria-autocomplete="list"
                  class="css-wmatm6-dummyInput-DummyInput"
                  id="react-select-2-input"
                  readonly=""
                  tabindex="0"
                  value=""
                />
              </div>
              <div
                class="CalculationOptions__indicators css-1hb7zxy-IndicatorsContainer"
              >
                <div
                  aria-hidden="true"
                  class="CalculationOptions__indicator CalculationOptions__clear-indicator css-13eygzs-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
                <span
                  class="CalculationOptions__indicator-separator css-43ykx9-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class="CalculationOptions__indicator CalculationOptions__dropdown-indicator css-1f9iddu-indicatorContainer"
                >
                  <i
                    class="CompassIcon icon-chevron-up ChevronUpIcon"
                  />
                </div>
              </div>
            </div>
            <div
              class="CalculationOptions__menu css-1rsmi4x-menu"
            >
              <div
                class="CalculationOptions__menu-list css-g29tl0-MenuList"
              >
                <div
                  class="KanbanCalculationOptions_CustomOption active"
                >
                  <span>
                    Count
                     
                  </span>
                </div>
                <div
                  class="KanbanCalculationOptions_CustomOption "
                >
                  <span>
                    Count Empty
                     
                    <i
                      class="CompassIcon icon-chevron-right ChevronRightIcon"
                    />
                  </span>
                </div>
                <div
                  class="KanbanCalculationOptions_CustomOption "
                >
                  <span>
                    Count Not Empty
                     
                    <i
                      class="CompassIcon icon-chevron-right ChevronRightIcon"
                    />
                  </span>
                </div>
                <div
                  class="KanbanCalculationOptions_CustomOption "
                >
                  <span>
                    Percent Empty
                     
                    <i
                      class="CompassIcon icon-chevron-right ChevronRightIcon"
                    />
                  </span>
                </div>
                <div
                  class="KanbanCalculationOptions_CustomOption "
                >
                  <span>
                    Percent Not Empty
                     
                    <i
                      class="CompassIcon icon-chevron-right ChevronRightIcon"
                    />
                  </span>
                </div>
                <div
                  class="KanbanCalculationOptions_CustomOption "
                >
                  <span>
                    Count Value
                     
                    <i
                      class="CompassIcon icon-chevron-right ChevronRightIcon"
                    />
                  </span>
                </div>
                <div
                  class="KanbanCalculationOptions_CustomOption "
                >
                  <span>
                    Count Unique Values
                     
                    <i
                      class="CompassIcon icon-chevron-right ChevronRightIcon"
                    />
                  </span>
                </div>
              </div>
            </div>
            <input
              name="calculation_options"
              type="hidden"
              value="count"
            />
          </div>
        </div>
        <div
          class="octo-spacer"
        />
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="Button IconButton"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
        <button
          class="Button IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-plus AddIcon"
          />
        </button>
      </div>
      <div
        class="octo-board-header-cell KanbanColumnHeader"
        draggable="true"
        style="opacity: 1;"
      >
        <span
          class="Label propColorBlue "
        >
          <input
            class="Editable undefined"
            placeholder="New Select"
            spellcheck="true"
            title="Q2"
            value="Q2"
          />
        </span>
        <div
          class="KanbanCalculation"
        >
          <button
            title="1"
            type="button"
          >
            <span>
              1
            </span>
          </button>
        </div>
        <div
          class="octo-spacer"
        />
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="Button IconButton"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
        <button
          class="Button IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-plus AddIcon"
          />
        </button>
      </div>
      <div
        class="octo-board-header-cell narrow"
      >
        Hidden columns
      </div>
      <div
        class="octo-board-header-cell narrow"
      >
        <button
          type="button"
        >
          <span>
            + Add a group
          </span>
        </button>
      </div>
    </div>
    <div
      class="octo-board-body"
      id="mainBoardBody"
    >
      <div
        class="octo-board-column"
      >
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <button
          type="button"
        >
          <span>
            + New
          </span>
        </button>
      </div>
      <div
        class="octo-board-column"
      >
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <button
          type="button"
        >
          <span>
            + New
          </span>
        </button>
      </div>
      <div
        class="octo-board-column narrow"
      >
        <div
          class="octo-board-hidden-item"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper"
            role="button"
          >
            <span
              class="Label propColorDefault "
            >
              Q3
            </span>
          </div>
          <button
            type="button"
          >
            <span>
              0
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`src/component/kanban/kanban should match snapshot 1`] = `
<div>
  <div
    class="Kanban"
  >
    <div
      class="octo-board-header"
      id="mainBoardHeader"
    >
      <div
        class="octo-board-header-cell KanbanColumnHeader"
        draggable="true"
        style="opacity: 1;"
      >
        <span
          class="Label propColorOrange "
        >
          <input
            class="Editable undefined"
            placeholder="New Select"
            spellcheck="true"
            title="Q1"
            value="Q1"
          />
        </span>
        <div
          class="KanbanCalculation"
        >
          <button
            title="2"
            type="button"
          >
            <span>
              2
            </span>
          </button>
        </div>
        <div
          class="octo-spacer"
        />
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="Button IconButton"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
        <button
          class="Button IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-plus AddIcon"
          />
        </button>
      </div>
      <div
        class="octo-board-header-cell KanbanColumnHeader"
        draggable="true"
        style="opacity: 1;"
      >
        <span
          class="Label propColorBlue "
        >
          <input
            class="Editable undefined"
            placeholder="New Select"
            spellcheck="true"
            title="Q2"
            value="Q2"
          />
        </span>
        <div
          class="KanbanCalculation"
        >
          <button
            title="1"
            type="button"
          >
            <span>
              1
            </span>
          </button>
        </div>
        <div
          class="octo-spacer"
        />
        <div
          aria-label="menuwrapper"
          class="MenuWrapper"
          role="button"
        >
          <button
            class="Button IconButton"
            type="button"
          >
            <i
              class="CompassIcon icon-dots-horizontal OptionsIcon"
            />
          </button>
        </div>
        <button
          class="Button IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-plus AddIcon"
          />
        </button>
      </div>
      <div
        class="octo-board-header-cell narrow"
      >
        Hidden columns
      </div>
      <div
        class="octo-board-header-cell narrow"
      >
        <button
          type="button"
        >
          <span>
            + Add a group
          </span>
        </button>
      </div>
    </div>
    <div
      class="octo-board-body"
      id="mainBoardBody"
    >
      <div
        class="octo-board-column"
      >
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <button
          type="button"
        >
          <span>
            + New
          </span>
        </button>
      </div>
      <div
        class="octo-board-column"
      >
        <div
          class="KanbanCard"
          draggable="true"
          style="opacity: 1;"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper optionsMenu"
            role="button"
          >
            <button
              class="Button IconButton"
              type="button"
            >
              <i
                class="CompassIcon icon-dots-horizontal OptionsIcon"
              />
            </button>
          </div>
          <div
            class="octo-icontitle"
          >
            <div
              class="octo-icon"
            >
              i
            </div>
            <div
              class="octo-titletext"
            >
              title
            </div>
          </div>
        </div>
        <button
          type="button"
        >
          <span>
            + New
          </span>
        </button>
      </div>
      <div
        class="octo-board-column narrow"
      >
        <div
          class="octo-board-hidden-item"
        >
          <div
            aria-label="menuwrapper"
            class="MenuWrapper"
            role="button"
          >
            <span
              class="Label propColorDefault "
            >
              Q3
            </span>
          </div>
          <button
            type="button"
          >
            <span>
              0
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
